### PR TITLE
Added internal support for Invoke-AzRestMethod in Test-AdfLinkedService

### LINKED ISSUE #144: Test-AdfLinkedService should have an option to use internally Invoke-AzRestMethod
**Is your feature request related to a problem? Please describe.**
Test-AdfLinkedService requires passing ClientID and ClientSecret. Usage of the Cmdlet requires the creation of a service principal and passing information into Cmdlet. 
 

**Describe the solution you'd like**
Underlying REST calls are calling endpoints of https://management.azure.com/ and they can be done automatically with Invoke-AzRestMethod that will use login information from the current Az context.

Therefore there should be an option to turn on the usage of Invoke-AzRestMethod with Test-AdfLinkedService.

**Describe alternatives you've considered**
When running on AzurePowershell task on Azure DevOps it could be possible to extract that information from service connection information. Requires extra for the caller

**Additional context**
-

### PR DESCRIPTION
**Pull request by jikuja**
Closes #144

### CODE REVIEW

**Comment by jikuja** [comment]
Please let me know if there are any changes needed before merging this. E.g. argument naming or logic of the arguments.

File: public/Test-AdfLinkedService.ps1 [hunk 1]
```diff
@@ -9,27 +9,37 @@ function Test-AdfLinkedService {
...
         [String] $TenantID,
-        [parameter(Mandatory = $true)] 
+        [parameter(Mandatory = $true, ParameterSetName="ClientDetails")] 
         [String] $ClientID,
-        [parameter(Mandatory = $true)] 
-        [String] $ClientSecret
+        [parameter(Mandatory = $true, ParameterSetName="ClientDetails")] 
+        [String] $ClientSecret,
+        [parameter(Mandatory = $true, ParameterSetName="AzRestMethod")]
+        [switch] $preferAzRestMethod
```
Comment (NowinskiK) [inline public/Test-AdfLinkedService.ps1:19]: I can't see the role of this switch

Comment (jikuja) [reply public/Test-AdfLinkedService.ps1:19]: It is used https://github.com/SQLPlayer/azure.datafactory.tools/pull/151/files#diff-ad136628efb917d1afaf8ff1157f12bd127c9cbbd4fed91fce0fa56a8885c83bR22 and here https://github.com/SQLPlayer/azure.datafactory.tools/pull/151/files#diff-ad136628efb917d1afaf8ff1157f12bd127c9cbbd4fed91fce0fa56a8885c83bR32

File: public/Test-AdfLinkedService.ps1 [hunk 2]
```diff
@@ -9,27 +9,37 @@ function Test-AdfLinkedService {
         [String] $ResourceGroupName,
         [parameter(Mandatory = $true)] 
         [String] $SubscriptionID,
```
Comment (NowinskiK) [inline public/Test-AdfLinkedService.ps1:11]: Using this opportunity, could you pass $SubscriptionID to calling functions appropriately?

File: public/Test-AdfLinkedService.ps1 (hunk 1 above)
Comment (NowinskiK) [inline public/Test-AdfLinkedService.ps1:19]: I can't see the role of this switch

**Review by NowinskiK — CHANGES_REQUESTED** [review]
Plus: Are you able to add a unit test for this function?

**Review by NowinskiK — CHANGES_REQUESTED** [review]
Could you please also add:
- example of usage in doc (readme file)
- unit test for the function

File: public/Test-AdfLinkedService.ps1 (hunk 1 above)
Comment (jikuja) [reply public/Test-AdfLinkedService.ps1:19]: Another option would be to use Az context of service principal details are not present in the command-line.

See other comments for more details.

**Pull request merged**